src/dmd/backend/obj.d

```diff
@@ -31,265 +31,159 @@ version (SPP)
-    int Obj_external_def(const(char)* );
-    int Obj_data_start(Symbol *sdata, targ_size_t datasize, int seg);
-    int Obj_external(Symbol *);
-    int Obj_common_block(Symbol *s, targ_size_t size, targ_size_t count);
-    int Obj_common_block(Symbol *s, int flag, targ_size_t size, targ_size_t count);
-    void Obj_lidata(int seg, targ_size_t offset, targ_size_t count);
-    void Obj_write_zeros(seg_data *pseg, targ_size_t count);
-    void Obj_write_byte(seg_data *pseg, uint _byte);
-    void Obj_write_bytes(seg_data *pseg, uint nbytes, void *p);
-    void Obj_byte(int seg, targ_size_t offset, uint _byte);
-    uint Obj_bytes(int seg, targ_size_t offset, uint nbytes, void *p);
-    void Obj_ledata(int seg, targ_size_t offset, targ_size_t data, uint lcfd, uint idx1, uint idx2);
-    void Obj_write_long(int seg, targ_size_t offset, uint data, uint lcfd, uint idx1, uint idx2);
-    void Obj_reftodatseg(int seg, targ_size_t offset, targ_size_t val, uint targetdatum, int flags);
-    void Obj_reftofarseg(int seg, targ_size_t offset, targ_size_t val, int farseg, int flags);
-    void Obj_reftocodeseg(int seg, targ_size_t offset, targ_size_t val);
-    int Obj_reftoident(int seg, targ_size_t offset, Symbol *s, targ_size_t val, int flags);
-    void Obj_far16thunk(Symbol *s);
-    void Obj_fltused();
-    int Obj_data_readonly(char *p, int len, int *pseg);
-    int Obj_data_readonly(char *p, int len);
-    int Obj_string_literal_segment(uint sz);
-    Symbol* Obj_sym_cdata(tym_t, char *, int);
-    void Obj_func_start(Symbol *sfunc);
-    void Obj_func_term(Symbol *sfunc);
-    void Obj_write_pointerRef(Symbol* s, uint off);
-    int Obj_jmpTableSegment(Symbol* s);
-
-    Symbol* Obj_tlv_bootstrap();
-
-    void Obj_gotref(Symbol *s);
-
-    uint Obj_addstr(Outbuffer *strtab, const(char)* );
-    Symbol* Obj_getGOTsym();
-    void Obj_refGOTsym();
-
-    version (OSX)
-    {
-        int Obj_getsegment(const(char)* sectname, const(char)* segname,
-                              int  _align, int flags);
-        void Obj_addrel(int seg, targ_size_t offset, Symbol *targsym,
-                           uint targseg, int rtype, int val = 0);
-    }
-    else
-    {
-        int Obj_getsegment(const(char)* name, const(char)* suffix,
-                              int type, int flags, int  _align);
-        void Obj_addrel(int seg, targ_size_t offset, uint type,
-                           uint symidx, targ_size_t val);
-        size_t Obj_writerel(int targseg, size_t offset, uint type,
-                               uint symidx, targ_size_t val);
-    }
+    public import stubobj;
 }
-
-version (OMFandMSCOFF)
+else
 {
+    /*******************************************
+     * Generic interface to the four object module file formats supported.
+     * Instead of using virtual functions (i.e. virtual dispatch) it uses
+     * static dispatch. Since config.objfmt never changes after initialization
```

Comment:
I don't think this is static dispatch, you're switching over a mutable global variable so it's still happening at run time.
Static dispatch would be if you put the back-end in templates and pass a specialized `Obj` final class / struct as a template parameter.

Reply:
It's static in that it relies on branch prediction.